### ISSUE TITLE
Urgent! Removable console warn

### ISSUE THREAD

**Issue by darklight9811** [formium/formik#3198]
### Current Behavior

If you use formik hooks outside of a formik form it will spam console warn on the application.

### Desired Behavior

Allow configurable console warn message to be optional.

### Suggested Solution

Add an option on the hook to disable the outside of form context warn log.

### Who does this impact? Who is this for?

In my company we have been using formik and since it's only tears, we made a wrap on it and solved most of the problems. But the only one we can't solve without changing the formik core, is remove the console warn message. We saw the need to use inputs outside of formik and making two different components for that wasn't a viable option.

### Describe alternatives you've considered

Remaking the useField hook and remove the console warn by hand.

### Additional context

Formik has been a hassle for a while, but since our application already has heavy dependency on it, we can't remove. We encapsulated code to prevent code repetition, rewrapped validation and logic into field logic and removed it from form context to allow easy forms. But that encapsulation became a problem when we had to use the components we created outside of the form. Since we already validate if it's in or not and it doesn't crash. The only thing that is killing perfomance is the console warn spam.

**Comment by johnrom** [comment]
I agree, errors and warnings should not be thrown in the hooks, they should be sent from components like Field which actually require the Context.
